esp/esdllib/esdl_script.cpp

```diff
@@ -234,6 +239,416 @@ interface IEsdlTransformOperationHttpHeader : public IInterface
+    {
+        ensureMysqlEmbeded();
+
+        m_name.set(stag.getValue("name"));
+        if (m_traceName.isEmpty())
+            m_traceName.set(m_name.str());
+
+        //select is optional, with select, a mysql call behaves like a for-each, binding and executing each iteration of the selected content
+        //without select, it executes once in the current context
+        m_select.setown(compileOptionalXpath(stag.getValue("select")));
+
+        m_vaultName.setown(compileOptionalXpath(stag.getValue("vault")));
+        m_secretName.setown(compileOptionalXpath(stag.getValue("secret")));
+        m_section.setown(compileOptionalXpath(stag.getValue("section")));
+        m_resultDelimiter.setown(compileOptionalXpath(stag.getValue("result-delimiter")));
+
+        m_server.setown(compileOptionalXpath(stag.getValue("server")));
+        m_user.setown(compileOptionalXpath(stag.getValue("user")));
+        m_password.setown(compileOptionalXpath(stag.getValue("password")));
+        m_database.setown(compileOptionalXpath(stag.getValue("database")));
+
+        //script can set any MYSQL options using an attribute with the same name as the option enum, for example
+        //    MYSQL_SET_CHARSET_NAME="'latin1'" or MYSQL_SET_CHARSET_NAME="$charset"
+        //
+        int attCount = stag.getLength();
+        for (int i=0; i<attCount; i++)
+        {
+            const char *attName = stag.getLocalName(i);
+            if (attName && strncmp("MYSQL_", attName, 6)==0)
+            {
+                Owned<ICompiledXpath> attXpath = compileOptionalXpath(stag.getValue(i));
+                if (attXpath)
+                {
+                    m_mysqlOptionNames.append(attName);
+                    m_mysqlOptionXpaths.append(*attXpath.getClear());
+                }
+            }
+        }
+
+        int type = 0;
+        while((type = xpp.next()) != XmlPullParser::END_DOCUMENT)
+        {
+            switch(type)
+            {
+                case XmlPullParser::START_TAG:
+                {
+                    StartTag stag;
+                    xpp.readStartTag(stag);
+                    const char *op = stag.getLocalName();
+                    if (isEmptyString(op))
+                        esdlOperationError(ESDL_SCRIPT_Error, m_tagname, "unknown error", m_traceName, !m_ignoreCodingErrors);
+                    if (streq(op, "bind"))
+                        m_parameters.append(*new CEsdlTransformOperationMySqlBindParmeter(xpp, stag, prefix));
+                    else if (streq(op, "sql"))
+                        readFullContent(xpp, m_sql);
+                    else
+                        xpp.skipSubTreeEx();
+                    break;
+                }
+                case XmlPullParser::END_TAG:
+                case XmlPullParser::END_DOCUMENT:
+                    return;
+            }
+        }
+
+        if (!m_section)
+            m_section.setown(compileXpath("'temporaries'"));
```

Comment:
As we begin to use the `temporaries` section more (tokens, mysql, variables) it is becoming more likely that a coder could create a collision. Perhaps soon we'll need to consider a way to prevent or at least warn of collisions.

Reply:
Agreed. Good point. At least for now there is an easy workaround, the script writer should guarantee a unique name.  But I'll improve it in a future version.  The actual tag name in the temporaries section shouldn't matter because they are not meant to be accessed from an xpath.  So in the future could be made unique.  I may block scripts from looking for things directly in the temporaries section anyway... it's meant for variable storage and the like.